rust/main/utils/run-locally/src/starknet/mod.rs

```diff
@@ -0,0 +1,587 @@
+
+    let addr_base = "0.0.0.0";
+    let default_config = StarknetConfig {
+        cli_path: katanad.clone(),
+
+        sierra_classes: sierra_classes.clone(),
+
+        node_addr_base: addr_base.to_string(),
+        node_port_base: 5050,
+
+        chain_id: "KATANA".to_string(),
+    };
+
+    let port_start = 5050u32;
+    let metrics_port_start = 9090u32;
+    let domain_start = 23448593u32;
+    let node_count = 2;
+
+    let nodes = (0..node_count)
+        .map(|i| {
+            (
+                launch_starknet_node(StarknetConfig {
+                    node_port_base: port_start + (i * 10),
+                    chain_id: format!("KATANA"),
+                    ..default_config.clone()
+                }),
+                format!("KATANA"),
+                metrics_port_start + i,
+                domain_start + i,
+            )
+        })
+        .collect::<Vec<_>>();
+
+    let deployer = "0xb3ff441a68610b30fd5e2abbf3a1548eb6ba6f3559f2862bf2dc757e5828ca"; // 1st katana account
+    let _linker = "validator";
+    let validator = &ValidatorConfig {
+        private_key: "0x0014d6672dcb4b77ca36a887e9a11cd9d637d5012468175829e9c6e770c61642"
+            .to_string(),
+        address: "0x00e29882a1fcba1e7e10cad46212257fea5c752a4f9b1b1ec683c503a2cf5c8a".to_string(),
+    };
+    let _relayer = "hpl-relayer";
+
+    sleep(Duration::from_secs(5));
+
+    let nodes = nodes
+        .into_iter()
+        .map(|v| (v.0.join(), v.1, v.2, v.3))
+        .map(|(launch_resp, chain_id, metrics_port, domain)| {
+            let mut starknet_cli = launch_resp.cli(&starklid);
+            starknet_cli.init(
+                STARKNET_KEYPAIR.into(),
+                STARKNET_ACCOUNT.into(),
+                KEYPAIR_PASSWORD.into(),
+                launch_resp.endpoint.rpc_addr.clone(),
+            );
+
+            let declarations =
+                utils::declare_all(starknet_cli.clone(), sierra_classes.clone()).join();
+
+            let deployments =
+                utils::deploy_all(starknet_cli, deployer.to_string(), declarations, domain);
+
+            (launch_resp, deployments, chain_id, metrics_port, domain)
+        })
+        .collect::<Vec<_>>();
+
+    // nodes with base deployments
+    let nodes = nodes
+        .into_iter()
+        .map(|v| (v.0, v.1.join(), v.2, v.3, v.4))
+        .map(|v| v.into())
+        .collect::<Vec<StarknetNetwork>>();
+
+    // TODO: what should be done here ?
+    // for (i, node) in nodes.iter().enumerate() {
+    //     let targets = &nodes[(i + 1)..];
+
+    //     if !targets.is_empty() {
+    //         println!(
+    //             "LINKING NODES: {} -> {:?}",
+    //             node.domain,
+    //             targets.iter().map(|v| v.domain).collect::<Vec<_>>()
+    //         );
+    //     }
+
+    //     for target in targets {
+    //         link_networks(&katanad, linker, validator, node, target);
+    //     }
+    // }
```

Comment:
in the cosmos implementation the `link_networks` step is used to configure the IGP oracles for each origin chain, and also configure the routing ISM which I'm not sure has coverage in the starknet e2e (but would very much like to have it covered if you're down). So the commented lines can be removed

Reply:
since this test is for starknet <-> starknet testing and there is no IGP support yet on starknet it doesnt make sense, so removed